src/scf/self_consistent_field.jl

```diff
@@ -36,7 +36,17 @@ function next_density(ham::Hamiltonian;
     eigres.converged || (@warn "Eigensolver not converged" iterations=eigres.iterations)
 
     # Update density from new ψ
-    occupation, εF = compute_occupation(ham.basis, eigres.λ; occupation_threshold)
+    if isnothing(ham.basis.model.fixed_fermi_level)
+        occupation, εF = compute_occupation(ham.basis, eigres.λ; occupation_threshold)
+    else
+        εF = ham.basis.model.fixed_fermi_level
+        occupation = compute_occupation(ham.basis, eigres.λ, εF)
+        minocc = maximum(minimum, occupation)
+        if ham.basis.model.temperature > 0 && minocc > occupation_threshold
+            @warn "One k-point has a high minimum occupation $minocc. You should probably " *
+                  "increase the number of bands."
+        end
+    end
```

Comment:
I'm not so happy to see all this bloat here plus it's repeated code. Maybe move to a separate function?